internal/xds/httpfilter/fault/fault.go

```diff
@@ -92,21 +92,30 @@ func parseConfig(cfg proto.Message) (httpfilter.FilterConfig, error) {
 	return config{config: msg}, nil
 }
 
-func (builder) ParseFilterConfig(cfg proto.Message) (httpfilter.FilterConfig, error) {
+func (provider) ParseFilterConfig(cfg proto.Message) (httpfilter.FilterConfig, error) {
 	return parseConfig(cfg)
 }
 
-func (builder) ParseFilterConfigOverride(override proto.Message) (httpfilter.FilterConfig, error) {
+func (provider) ParseFilterConfigOverride(override proto.Message) (httpfilter.FilterConfig, error) {
 	return parseConfig(override)
 }
 
-func (builder) IsTerminal() bool {
-	return false
+func (provider) IsTerminal() bool { return false }
+func (provider) IsClient() bool   { return true }
+func (provider) IsServer() bool   { return false }
```

Comment:
As discussed it would be nice if we could find a way to use the type system to avoid potential for disagreement between these functions and the filter that is produced's behavior, but this is OK for now.